Use prefix instead of different ports in redis database - incomplete

A-atmos opened the pull request:
For Issue: #207 
### This PR isn't ready to merge, but is here to ask for recommendation

This PR passes the database tests, but `alert.log` isn't created, which result in failing of other integration tests!

A-atmos committed:
Modified redis manager to use prefix as id instead of port

A-atmos committed:
Modified redis manager to use prefix as id instead of port

A-atmos committed:
Modified to use id as prefix instead of port

A-atmos committed:
Modified to use id as prefix instead of port

A-atmos committed:
Changed database to publish and store keys with prefix

A-atmos committed:
Changed database to publish and store keys with prefix

A-atmos committed:
Changed modules to use prefix instead of port

A-atmos committed:
Changed modules to use prefix instead of port

A-atmos committed:
Remove keys from the default database instead of killing the server

A-atmos committed:
Remove keys from the default database instead of killing the server

A-atmos committed:
Argument to use custom prefix instead of self generated

A-atmos committed:
Argument to use custom prefix instead of self generated

A-atmos committed:
Set prefix on database before processing

A-atmos committed:
Set prefix on database before processing

A-atmos committed:
Modified database tests to use prefix

A-atmos committed:
Modified database tests to use prefix

AlyaGomaa committed:
Merge remote-tracking branch 'A-atmos/enhance-redis-database' into a-atmos-enhance-redis-database

# Conflicts:
#	slips_files/core/database/_profile_flow.py
#	slips_files/core/database/database.py

AlyaGomaa commented:
hey @A-atmos i just rebased your branch to the latest develop

the issue here is that no msgs are received in any channel because of the is_msg_intended_for() function in slips_utils
this function doesn't have the prefix when checking for msgs
if i were you i'd move it the db file, fix all modules accordingly, and change the function to use the prefix 

thank you for the effort this is a really tough issue!

A-atmos commented:
Thank you for the appreciation. I'm doing my best to resolve this issue, but I'm still facing some issues in the integration tests. I moved the `utils.is_msg_intended_for()` in `database.py` and refactored the code accordingly. I'm not getting any errors while testing database and other individual tests, but while testing the integration tests, I'm still facing some issues. But I'm trying to fix them. I'm get this issue resolved as soon as possible. Thanks again!

AlyaGomaa commented:
hey @A-atmos integration tests run on files in our dataset/, try to see which file is failing in the test and then run slips normally on that file and check what errors it gives you
also you may need to modify the database fixture in conftest.py to match the new changes in the db

A-atmos committed:
Changed the is_msg_intended_for

A-atmos committed:
Refactored modules according to the changes made

A-atmos committed:
Merge branch 'develop' into enhance-redis-database

A-atmos committed:
Resolved error due to merge conflict

A-atmos committed:
Resolved errors in ARP
